lib/User_LDAP.php

```diff
@@ -423,15 +302,69 @@ public function getBackendName(){
 
 	/**
 	 * @param $uid
-	 * returns false if no user wes found. Will lead to the proxy to check the next server
-	 * @return string[]
+	 * @return string[]|false false if user user was not found
 	 */
 	public function getSearchTerms($uid) {
-		$user = $this->access->userManager->get($uid);
-		if ($user === null) {
-			return [];
+		$userEntry = $this->userManager->getByOwnCloudUID($uid);
+		if (!$userEntry) {
+			return false;
 		}
-		return $user->getSearchTerms();
+		return $userEntry->getSearchTerms();
 	}
 
+
+	/**
+	 * Get a users email address
+	 *
+	 * @param string $uid The username
+	 * @return string|null|false false if user was not found, null if no email is set
+	 * @since 10.0
+	 */
+	public function getEMailAddress($uid) {
+		$userEntry = $this->userManager->getByOwnCloudUID($uid);
+		if (!$userEntry) {
+			return false;
+		}
+		return $userEntry->getEMailAddress();
+	}
+	/**
+	 * Get a users email address
+	 *
+	 * @param string $uid The username
+	 * @return string|null|false false if user was not found, null if no email is set
+	 * @since 10.0
+	 */
+	public function getQuota($uid) {
+		$userEntry = $this->userManager->getByOwnCloudUID($uid);
+		if (!$userEntry) {
+			return false;
+		}
+		return $userEntry->getQuota();
+	}
+	/**
+	 * Get avatar for a users account for core powered user search
+	 *
+	 * FIXME This is an expensive operation and takes roughly half a second to parse the data and create the image. This might be too slow for sync jobs.
```

Comment:
Side question for the future: is it possible to store the raw image data (I think it's base64 encoded) and compare the data we have and the data we receive, and then create the image if the data is different? I'm not sure if it will improve the performance, but it might be worthy to have a try.